plugins/outputs/iotdb/README.md

```diff
@@ -0,0 +1,113 @@
+# IoTDB Output Plugin
+
+This output plugin saves Telegraf metrics to an Apache IoTDB backend,
+supporting session connection and data insertion.
+
+## Apache IoTDB
+
+Apache IoTDB (Database for Internet of Things) is an IoT native database with
+high performance for data management and analysis, deployable on the edge and
+the cloud. Due to its light-weight architecture, high performance and rich
+feature set together with its deep integration with Apache Hadoop, Spark and
+Flink, Apache IoTDB can meet the requirements of massive data storage,
+high-speed data ingestion and complex data analysis in the IoT industrial
+fields.
+
+For more details consult the [Apache IoTDB website](https://iotdb.apache.org)
+or the [Apache IoTDB GitHub page](https://github.com/apache/iotdb).
+
+## Getting started
+
+Before using this plugin, please configure the IP address, port number,
+user name, password and other information of the database server,
+as well as some data type conversion, time unit and other configurations.
+
+Please see the [configuration section](#Configuration) for an example
+configuration.
+
+## Please pay attention to these points
+
+1. IoTDB currently **DOES NOT support unsigned integer**.
+There are three available options of converting uint64, which are specified by
+parameter `uint64_conversion`.
+
+   - `int64_clip`, default option. If an unsigned integer is greater than
+   `math.MaxInt64`, save it as `int64`; else save `math.MaxInt64`
+   (9223372036854775807).
+   - `int64`, force converting an unsigned integer to a`int64`,no mater
+   what the value it is. This option may lead to exception if the value is
+   greater than `int64`.
+   - `text`force converting an unsigned integer to a string, no mater what the
+   value it is.
+
+2. IoTDB supports a variety of time precision. You can specify which precision
+you want using the `timestamp_precision` option. Default is `nanosecond`.
+Other options are "second", "millisecond", "microsecond".
+
+3. Till now, IoTDB can not support Tag indexing well. To see current process
+   method, please refer to [InfluxDB-Protocol Adapter](
+   https://iotdb.apache.org/UserGuide/Master/API/InfluxDB-Protocol.html).
+   There are two available options of converting tags, which are specified by
+   parameter `convert_tags_to`:
+
+   - `fields`. Treat Tags as measurements. For each Key:Value in Tag,
+   convert them into Measurement, Value, DataType, which are supported in IoTDB.
+   - `device_id`, default option. Treat Tags as part of device id. Tags
+   is subtree of 'Name'.
+
+   For example, there is a metric:
+
+   `Name="root.sg.device", Tags={tag1="private", tag2="working"}, Fields={s1=100, s2="hello"}`
+
+   - `fields`, result: `root.sg.device, s1=100, s2="hello", tag1="private", tag2="working"`
+   - `device_id`, result: `root.sg.device.private.working, s1=100, s2="hello"`
```

Comment:
Let's use the heading here to describe what it contains. (We would also hope people are paying attention to the whole document, not just these points :smiley:)

The items in the ordered list are large enough that they would work better as subheadings.

You use the terms "parameter" or "option" here but telegraf docs call it "setting".

```suggestion
## Metric Translation
IoTDB uses a different data format for metric data than telegraf. It is important to note that depending on the metrics being written, the translation may be lossy. This plugin translates to IoTDB format in the following ways:

### Unsigned Integers
IoTDB currently **DOES NOT support unsigned integer**.
There are three available options of converting uint64, which are specified by
setting `uint64_conversion`.

   - `int64_clip`, default option. If an unsigned integer is greater than
   `math.MaxInt64`, save it as `int64`; else save `math.MaxInt64`
   (9223372036854775807).
   - `int64`, force converting an unsigned integer to a`int64`,no mater
   what the value it is. This option may lead to exception if the value is
   greater than `int64`.
   - `text`force converting an unsigned integer to a string, no mater what the
   value it is.

### Time Precision
IoTDB supports a variety of time precision. You can specify which precision
you want using the `timestamp_precision` setting. Default is `nanosecond`.
Other options are "second", "millisecond", "microsecond".

### Metadata (tags)
IoTDB uses a tree model for metadata while Telegraf uses a tag model.
(See [InfluxDB-Protocol Adapter](
https://iotdb.apache.org/UserGuide/Master/API/InfluxDB-Protocol.html)
There are two available options of converting tags, which are specified by
setting `convert_tags_to`:

   - `fields`. Treat Tags as measurements. For each Key:Value in Tag,
   convert them into Measurement, Value, DataType, which are supported in IoTDB.
   - `device_id`, default option. Treat Tags as part of device id. Tags
   is subtree of 'Name'.

   For example, there is a metric:

   `Name="root.sg.device", Tags={tag1="private", tag2="working"}, Fields={s1=100, s2="hello"}`

   - `fields`, result: `root.sg.device, s1=100, s2="hello", tag1="private", tag2="working"`
   - `device_id`, result: `root.sg.device.private.working, s1=100, s2="hello"`
```

Reply:
Thanks for suggestions! `README.md` updated. Relevant comments in `sample.conf` are also updated.